src/main/pages/che-7/administration-guide/assembly_deploy-kubernetes-image-puller-on-openshift.adoc

```diff
@@ -0,0 +1,108 @@
+deployment.apps/kubernetes-image-puller created
+configmap/k8s-image-puller created
+role.rbac.authorization.k8s.io/create-daemonset created
+rolebinding.rbac.authorization.k8s.io/create-daemonset created
+serviceaccount/k8s-image-puller created
+```
+
+To set different values than the defaults, use `oc process -p parameterName=value -f deploy/openshift/$template | oc apply -f -`
+
+=== Verify Installation
+After installation, you should see a deployment, and a `kubernetes-image-puller` deployment, and daemonset named `DAEMONSET_NAME` with a pod for each node in your cluster:
+
+```
+$ kubectl get deployment,daemonset,pod -n k8s-image-puller
+deployment.extensions/kubernetes-image-puller   1/1       1            1           2m19s
+
+NAME                                           DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
+daemonset.extensions/kubernetes-image-puller   1         1         1         1            1           <none>          2m10s
+
+NAME                                           READY     STATUS    RESTARTS   AGE
+pod/kubernetes-image-puller-5495f46497-mkd4p   1/1       Running   0          2m18s
+pod/kubernetes-image-puller-n8bmf              3/3       Running   0          2m10s
+```
+
+Check that the `ConfigMap` named `k8s-image-puller` has the values you specified in your parameter substitution, or that they contain the default values:
+
+```
+$ kubectl get configmap k8s-image-puller -o yaml
+apiVersion: v1
+data:
+  CACHING_INTERVAL_HOURS: "1"
+  CACHING_MEMORY_LIMIT: 20Mi
+  CACHING_MEMORY_REQUEST: 10Mi
+  DAEMONSET_NAME: kubernetes-image-puller
+  IMAGES: |
+    java11-maven=quay.io/eclipse/che-java11-maven:nightly; che-theia=eclipse/che-theia:next; java-plugin-runner=eclipse/che-remote-plugin-runner-java8:latest;
+  NAMESPACE: k8s-image-puller
+  NODE_SELECTOR: '{}'
+kind: ConfigMap
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"v1","data":{"CACHING_INTERVAL_HOURS":"1","CACHING_MEMORY_LIMIT":"20Mi","CACHING_MEMORY_REQUEST":"10Mi","DAEMONSET_NAME":"kubernetes-image-puller","IMAGES":"java11-maven=quay.io/eclipse/che-java11-maven:nightly; che-th
+eia=eclipse/che-theia:next; java-plugin-runner=eclipse/che-remote-plugin-runner-java8:latest;\n","NAMESPACE":"k8s-image-puller","NODE_SELECTOR":"{}"},"kind":"ConfigMap","metadata":{"annotations":{},"name":"k8s-image-puller","namespace":"k
+8s-image-puller"},"type":"Opaque"}
+  creationTimestamp: 2020-02-17T22:40:13Z
+  name: k8s-image-puller
+  namespace: k8s-image-puller
+  resourceVersion: "72250"
+  selfLink: /api/v1/namespaces/k8s-image-puller/configmaps/k8s-image-puller
+  uid: 76430ed6-51d6-11ea-9c19-52fdfc072182
+```
+
+When creating a workspace that relies on cached images specified in `IMAGES`, you should see Eclipse Che confirm that the images are already present when the workspace is starting:
+
+image::installation/image-already-present-on-machine.png[]
```

Comment:
This message will not necessarily be logged, depending on cluster configuration. Some clusters require a `PullPolicy` of `Always`, and in that case, you will still get the "getting image" logline. In that case the cluster has to get the layer digests from the registry and verify that those layers are present, in order to prevent users running images they do not have access to.